test/jdk/sun/tools/jhsdb/JShellHeapDumpTest.java

```diff
@@ -91,12 +95,16 @@ public static void launch(String expectedMessage, List<String> toolArgs)
                 System.out.println("Jshell not alive");
             }
         }
+        return true;
     }
 
     public static void launch(String expectedMessage, String... toolArgs)
         throws IOException {
 
-        launch(expectedMessage, Arrays.asList(toolArgs));
+        boolean res = launch(expectedMessage, Arrays.asList(toolArgs), true);
+        if (!res && !doSleep) {
```

Comment:
You should explain why we allow a retry here, and only for !doSleep, like we do in printStackTraces(). Note the doSleep reason is a bit different here. It is because the sleep allows the debuggee to stabilize, making it very unlikely that jmap will fail.